setup/git/000_EnvSetup.rb

```diff
@@ -85,60 +59,68 @@
 action=nocheck
 # Install to the default base directory.
 basedir=default
-END
-    on host, 'pkgadd -a /root/shutupsolaris -d http://get.opencsw.org/now all'
-    on host, '/opt/csw/bin/pkgutil -U all'
-    on host, '/opt/csw/bin/pkgutil -i -y ruby19_dev'
-    on host, '/opt/csw/bin/pkgutil -i -y ruby19'
-    on host, 'ln -sf /opt/csw/bin/gem19 /usr/bin/gem'
-    on host, 'ln -sf /opt/csw/bin/ruby19 /usr/bin/ruby'
+      END
+      on host, 'pkgadd -d http://get.opencsw.org/now -a /root/shutupsolaris -n all'
+      on host, '/opt/csw/bin/pkgutil -U all'
+      on host, '/opt/csw/bin/pkgutil -y -i git'
+      on host, '/opt/csw/bin/pkgutil -y -i gtar'
+    end
+    host.add_env_var('PATH', '/opt/csw/bin')
   end
 end
 
 install_packages_on(hosts, PACKAGES, :check_if_exists => true)
 
-hosts.each do |host|
-  case host['platform']
-  when /windows/
-    arch = host[:ruby_arch] || 'x86'
-    step "#{host} Selected architecture #{arch}"
+step "Install puppet-runtime" do
+  step 'grab the latest runtime tag'
+  runtime_dir = Dir.mktmpdir('puppet-runtime')
+  `git clone --depth 1 git@github.com:puppetlabs/puppet-runtime.git #{runtime_dir}`
+  runtime_tag = nil
+  Dir.chdir runtime_dir do
+    runtime_tag = `git describe --abbrev=0`.chomp
```

Comment:
Might want to use `--first-parent` here as well. There's been some discussion about merge-ups from older branches causing issues with `git-describe`, where describe will pick up the "older" branch because it happens "first" in the history (Eg: Getting a `5.5.x` related tag on the `master` branch).

```
       --first-parent
           Follow only the first parent commit upon seeing a merge commit. This is
           useful when you wish to not match tags on branches merged in the
           history of the target commit.
```

Reply:
`--first-parent` doesn't really get me what I want here. I'm looking for the last tag on the branch
```
[0] Melissa@bubba:~/beaker-puppet:(maint/master/git-based-acceptance)$ git describe
0.16.0-2-g078284b
[0] Melissa@bubba:~/beaker-puppet:(maint/master/git-based-acceptance)$ git describe --first-parent
0.16.0-2-g078284b
[0] Melissa@bubba:~/beaker-puppet:(maint/master/git-based-acceptance)$ git describe --abbrev=0
0.16.0
```

So I want 0.16.0 here, without the `-2-g078284b`

Reply:
I think you want both `--abbrev` and `--first-parent`.

Reply:
ooooh, yeah, that makes more sense. Updated!